llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -348,6 +348,42 @@ Value *VPInstruction::generateInstruction(VPTransformState &State,
     Value *Zero = ConstantInt::get(ScalarTC->getType(), 0);
     return Builder.CreateSelect(Cmp, Sub, Zero);
   }
+  case VPInstruction::ExplicitVectorLength: {
+    // Compute EVL
+    auto GetSetVL = [=](VPTransformState &State, Value *EVL) {
+      assert(EVL->getType()->isIntegerTy() &&
+             "Requested vector length should be an integer.");
+
+      // TODO: Add support for MaxSafeDist for correct loop emission.
+      Value *VFArg = State.Builder.getInt32(State.VF.getKnownMinValue());
+
+      Value *GVL = State.Builder.CreateIntrinsic(
+          State.Builder.getInt32Ty(), Intrinsic::experimental_get_vector_length,
+          {EVL, VFArg, State.Builder.getTrue()});
+      return GVL;
+    };
+    // TODO: Restructure this code with an explicit remainder loop, vsetvli can
+    // be outside of the main loop.
+    assert(Part == 0 && "No unrolling expected for predicated vectorization.");
+    // Compute VTC - IV as the EVL(requested vector length).
+    Value *Index = State.get(getOperand(0), 0);
+    Value *TripCount = State.get(getOperand(1), VPIteration(0, 0));
+    Value *EVL = State.Builder.CreateSub(TripCount, Index);
```

Comment:
> Suggestion to use`ApplicationVectorLength` (from  https://reviews.llvm.org/D99750?id=558054#inline-1552142)

Reply:
AS I said before, ApplicationVectorLength is just an input argument of the get_vector_length instinsic, while EVL - is the result, returned by this intrinsic. It is used in all LLVM documents with this meaning.

Reply:
So we're in agreement that `TC - Index`, computed here and fed as an input argument to the get_vector_length instinsic, is in fact the Application VL to be named `AVL`, and the result returned by `GetSetVL()` should be named `EVL`, right? Rather than `EVL` and `SetVL`, respectively.

Reply:
Yes